src/sharding/ShardClientUtil.js

```diff
@@ -128,28 +128,28 @@ class ShardClientUtil {
   /**
    * Evaluates a script or function on all shards, or a given shard, in the context of the {@link Client}s.
    * @param {string|Function} script JavaScript to run on each shard
-   * @param {number} [shard] Shard to run script on, all if undefined
+   * @param {BroadcastEvalOptions} [options={}] The options for the broadcast
    * @returns {Promise<*>|Promise<Array<*>>} Results of the script execution
    * @example
    * client.shard.broadcastEval('this.guilds.cache.size')
    *   .then(results => console.log(`${results.reduce((prev, val) => prev + val, 0)} total guilds`))
    *   .catch(console.error);
    * @see {@link ShardingManager#broadcastEval}
    */
-  broadcastEval(script, shard) {
+  broadcastEval(script, options = {}) {
     return new Promise((resolve, reject) => {
       const parent = this.parentPort || process;
-      script = typeof script === 'function' ? `(${script})(this)` : script;
+      script = typeof script === 'function' ? `(${script})(this, ${JSON.stringify(options.context ?? {})})` : script;
```

Comment:
`null` and `undefined` (both being falsy values) defaulting to an empty object (being truthy) might cause some headaches in the developer's side, and `JSON.stringify` allows both as a value, so why not let them be passed?
```suggestion
      script = typeof script === 'function' ? `(${script})(this, ${JSON.stringify(options.context)})` : script;
```